core/src/main/java/org/infinispan/remoting/transport/Transport.java

```diff
@@ -178,4 +181,23 @@
     */
    Set<String> getSitesView();
 
+   <T> CompletionStage<T> invokeCommand(Address target, ReplicableCommand command,
+                                        ResponseCollector<T> collector, DeliverOrder deliverOrder,
+                                        long timeout, TimeUnit unit);
+
+   <T> CompletionStage<T> invokeCommand(Collection<Address> targets, ReplicableCommand command,
```

Comment:
Actually I'd want the ResponseCollector-based RpcManager API to use a `CompletionStage` as well.

For `BaseAsyncInterceptor.asyncValue()` yes, it was more convenient to assume a `CompletableFuture`, but a) there's always `CompletionStage.toCompletableFuture()` and b) we may want to use our own `CompletionStage` implementation in the future, and `asyncValue()` could be extended to deal with that implementation.